lib/token.js

```diff
@@ -578,7 +608,33 @@ function parseFromUrl(sdk, url) {
         // Remove the hash from the url
         removeHash(sdk);
       }
-      return handleOAuthResponse(sdk, oauthParams, res, urls);
+      return handleOAuthResponse(sdk, oauthParams, res, urls)
+      .then(function(res) {
+        if (oauthParams.responseType !== 'code') {
+          return res;
+        }
+
+        // PKCE authorization_code flow
+        // Retreive saved values and build oauthParans for call to /token
+        var meta = sdk.pkce.loadMeta();
+        var clonedParams = util.clone(oauthParams) || {};
+        util.extend(clonedParams, {
+          grantType: 'authorization_code',
+          authorizationCode: res,
+          codeVerifier: meta.codeVerifier,
+          responseType: meta.responseType,
+          redirectUri: meta.redirectUri
+        });
+        delete clonedParams.state;
```

Comment:
See on Line 144 where options.state is compared with response.state. state here is left over from /authorize call and was validated the first time through handleAuthResponse()

Reply:
Unfortunately this would be a breaking change.

The `state` param is known primarily for mitigating CSRF attacks, however another common use-case is to use it for caching an end-user's state in an application.

Consider the following:

1. User navigates to `example.com/protected`
2. Since User is not authenticated, they're redirected to `example.com/login`
3. Once authenticated, the user is redirected to the registered callback URI, say `example.com/cb`

Our OIDC integrations bounce the user back to `/protected` since we control the login and redirect handlers, however customers using AuthJS may not be handling the previous state themselves, and rely on this cookie.

Reply:
I believe this is being validated in the response from /authorize The /token endpoint does not send back any state and would cause an error if we attempted to validate it.